File: mp2i/files/C2/nbchiffres.c
#include <stdio.h>
#include <assert.h>

int nb_chiffres(int n)
{
    assert(n >= 0);
    int p = 0;
    if (n == 0)
    {
        return 1;
    }
    while (n > 0)
    {
        n = n / 10;
        p = p + 1;
    }
    return p;
}

int main()
{
    assert (nb_chiffres(0)==1);
    assert (nb_chiffres(7)==1);
    assert (nb_chiffres(10)==2);
    assert (nb_chiffres(11)==2);
    assert (nb_chiffres(2567)==4);
}

File: itc/files/C10/nbchiffres.py
def nbchiffres(n:int)->int:
    '''Renvoie le nombres de chiffres de n en base 10'''
    assert n>=0, "Le nombre doit être positif"
    if n==0:
        return 1
    p = 0
    while n>0:
        p = p + 1
        n = n//10
    return p
